webpack/assets/javascripts/react_app/actions/NotificationActions.js

```diff
@@ -1,18 +1,27 @@
 import API from '../API';
 import AppDispatcher from '../dispatcher';
-import {ACTIONS} from '../constants';
+import NotificationsStore from '../stores/NotificationsStore';
+import { ACTIONS, STATUS } from '../constants';
 
 const TIMER = 10000;
 
 export default {
   getNotifications(url) {
-    if (document.visibilityState === 'visible') {
+    if (document.visibilityState !== 'hidden' &&
```

Comment:
unloaded means we navigate to another page right? maybe we should also check if turbolinks is not active (e.g. I clicked on a really slow page to load that is going to refresh the entire page, no point in calling notifications as we'll do it again for the new page?)